components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java

```diff
@@ -1185,7 +1186,8 @@ public Response applicationsApplicationIdOauthKeysKeyMappingIdGenerateTokenPost(
                     try {
                         // verify that the provided jsonInput is a valid json
                         if (body.getAdditionalProperties() != null
-                                && !body.getAdditionalProperties().toString().isEmpty()) {
+                                && !body.getAdditionalProperties().toString().isEmpty()
+                                && !Objects.equals(body.getAdditionalProperties().toString(), "{}")) {
```

Comment:
Any particular reason for checking whether there aren't any properties by comparing the string value of the object returned? Feel like that'd be a very error prone way of comparing (as it seems to have been the case in this issue you're addressing). 

Isn't what's returned by the `getAdditionalProperties()` method some sort of a collection? If it's possible for different kinds of collections to be returned here, wouldn't it be better to handle this `isEmpty` check for each of those possible types instead of doing a string comparison?